apps/root/src/pages/transfer/transfer-container/components/recipient-address/index.tsx

```diff
@@ -0,0 +1,122 @@
+  const activeWallet = useActiveWallet();
+  const [recipientInput, setRecipientInput] = React.useState<string>('');
+  const { token, recipient: storedRecipient } = useTransferState();
+  const currentNetwork = useCurrentNetwork();
+  const [shouldShowContactList, setShouldShowContactList] = React.useState<boolean>(false);
+
+  const inputRegex = RegExp(/^[A-Fa-f0-9x]*$/);
+  const { isValidRecipient } = validateAddress(recipientInput, activeWallet?.address);
+
+  React.useEffect(() => {
+    const { isValidRecipient: isValid } = validateAddress(recipientParam || '', activeWallet?.address);
+    if (recipientParam && isValid) {
+      setRecipientInput(recipientParam);
+      dispatch(setRecipient(recipientParam));
+    }
+  }, [dispatch, validateAddress]);
+
+  const onRecipientChange = (nextValue: string) => {
+    if (!inputRegex.test(nextValue)) {
+      return;
+    }
+    setRecipientInput(nextValue);
+    const { isValidRecipient: isValid } = validateAddress(nextValue, activeWallet?.address);
+
+    if (isValid) {
+      dispatch(setRecipient(nextValue));
+      if (token) {
+        replaceHistory(`/transfer/${currentNetwork.chainId}/${token.address}/${nextValue}`);
+      }
+    } else if (storedRecipient) {
+      // Clear the stored recipient if the input is not valid
+      dispatch(setRecipient(''));
+    }
+  };
+
+  const onClickContact = (contactAddress: string) => {
+    onRecipientChange(contactAddress);
+    setShouldShowContactList(false);
+  };
+
+  const hasError = recipientInput !== '' && recipientInput !== null && !isValidRecipient;
+
+  let error = '';
+  if (hasError) {
+    if (recipientInput.toLowerCase() === activeWallet?.address.toLowerCase()) {
+      error = intl.formatMessage({
+        defaultMessage: 'Transfer address cannot be the same as your address',
+        description: 'errorSameAddress',
+      });
+    } else {
+      error = intl.formatMessage({
+        defaultMessage: 'This is not a valid address',
+        description: 'errorInvalidAddress',
+      });
+    }
+  }
+
+  return (
+    <>
+      <ContactListModal
+        shouldShow={shouldShowContactList}
+        setShouldShow={setShouldShowContactList}
+        onClickContact={onClickContact}
+      />
+      <TextField
+        id="recipientAddress"
+        value={recipientInput}
+        placeholder={intl.formatMessage(
+          defineMessage({
+            defaultMessage: 'Recipient Address',
+            description: 'recipientAddress',
+          })
+        )}
+        autoComplete="off"
+        autoCorrect="off"
+        error={hasError}
+        helperText={hasError ? error : ''}
+        fullWidth
+        type="text"
+        margin="normal"
+        spellCheck="false"
+        onChange={(evt) => onRecipientChange(evt.target.value)}
+        inputProps={{
+          pattern: '^0x[A-Fa-f0-9]*$',
+          minLength: 1,
+          maxLength: 79,
+        }}
+        InputProps={{
+          endAdornment: (
```

Comment:
Could be cool to also add a paste button here, I think the transfer-to-modal has it